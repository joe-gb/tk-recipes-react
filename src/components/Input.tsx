import React from "react";
import styled from 'styled-components';
import { Text  } from "./Text"


export const Label = styled.label`
  font-size: 14px;
  font-weight: 400;
  line-height: 24px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
  margin: 20px auto;`

interface InputProps<T> extends React.InputHTMLAttributes<T>{
    error?: String | null,
    label?: String,
}

const InputError = styled(Text)`
  color: #EA3E3E;
  font-size: 12px;
`

const WrappedInput = ({className, error, label, ...props}: InputProps<any>) => (
    <div className={className}>
        {label &&
            <Label htmlFor={className}>{label}</Label>
        }
        <div>
            <input id={className} {...props} />
        </div>
        {error &&
            <InputError>{error}</InputError>
        }

    </div>
)

export const Input = styled(WrappedInput)`
  & {
    margin: 10px auto;
  }
  & div {
    align-items: center;
    width: 100%;
    min-height: 40px;
    z-index: initial;
    font-size: 16px;
    background: #FFFFFF;
    caret-color: auto;
    border: 1px solid ${props => props.error ? "#EA3E3E" : "#C9D3DF"};
    border-top-left-radius: 4px;
    border-bottom-left-radius: 4px;
    border-top-right-radius: 4px;
    border-bottom-right-radius: 4px;
    transition: opacity 0.2s ease-in-out, border-color 0.2s ease-in-out;

    & div:hover {
      border-left: 1px solid #738094;
      border-right: 1px solid #738094;
      border-top: 1px solid #738094;
      border-bottom: 1px solid #738094;
    }

    & input {      
      height: 40px;
      box-sizing: border-box;
      width: 100%;
      padding: 0 14px;
      border-radius: 2px;
      line-height: 32px;
      font-size: 16px;
      border: none;
      color: #323B4D;
    }

    & input:focus {
      outline: none;
    }

  }

`;