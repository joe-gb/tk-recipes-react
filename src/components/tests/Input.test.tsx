import React from "react";
import { render, screen } from "@testing-library/react";
import { Input } from "../Input";

test("Snapshot test with label", () => {
  let { container } = render(<Input label="Enter your answer" />);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("Enter your answer");
  expect(screen.getByLabelText("Enter your answer")).toBeDefined();
});

test("Snapshot test plain label", () => {
  let { container } = render(<Input />);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("");
});

test("Snapshot test with placeholder", () => {
  let { container } = render(<Input placeholder="Type here" />);
  expect(container).toMatchSnapshot();
  expect(container.textContent).toBe("");
  expect(screen.getByPlaceholderText("Type here")).toBeDefined();
});
