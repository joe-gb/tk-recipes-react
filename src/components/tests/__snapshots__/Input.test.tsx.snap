// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test plain label 1`] = `
.c0 {
  margin: 10px auto;
}

.c0 div {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  min-height: 40px;
  z-index: initial;
  font-size: 16px;
  background: #ffffff;
  caret-color: auto;
  border: 1px solid #C9D3DF;
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  -webkit-transition: opacity 0.2s ease-in-out,border-color 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out,border-color 0.2s ease-in-out;
}

.c0 div div:hover {
  border-left: 1px solid #738094;
  border-right: 1px solid #738094;
  border-top: 1px solid #738094;
  border-bottom: 1px solid #738094;
}

.c0 div input {
  height: 40px;
  box-sizing: border-box;
  width: 100%;
  padding: 0 14px;
  border-radius: 2px;
  line-height: 32px;
  font-size: 16px;
  border: none;
  color: #323b4d;
}

.c0 div input:focus {
  outline: none;
}

<div>
  <div
    class="c0"
  >
    <div>
      <input
        id="sc-iCfMLu c0"
      />
    </div>
  </div>
</div>
`;

exports[`Snapshot test with label 1`] = `
.c1 {
  font-size: 14px;
  font-weight: 400;
  line-height: 24px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
  margin: 20px auto;
}

.c0 {
  margin: 10px auto;
}

.c0 div {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  min-height: 40px;
  z-index: initial;
  font-size: 16px;
  background: #ffffff;
  caret-color: auto;
  border: 1px solid #C9D3DF;
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  -webkit-transition: opacity 0.2s ease-in-out,border-color 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out,border-color 0.2s ease-in-out;
}

.c0 div div:hover {
  border-left: 1px solid #738094;
  border-right: 1px solid #738094;
  border-top: 1px solid #738094;
  border-bottom: 1px solid #738094;
}

.c0 div input {
  height: 40px;
  box-sizing: border-box;
  width: 100%;
  padding: 0 14px;
  border-radius: 2px;
  line-height: 32px;
  font-size: 16px;
  border: none;
  color: #323b4d;
}

.c0 div input:focus {
  outline: none;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
      for="sc-iCfMLu c0"
    >
      Enter your answer
    </label>
    <div>
      <input
        id="sc-iCfMLu c0"
      />
    </div>
  </div>
</div>
`;

exports[`Snapshot test with placeholder 1`] = `
.c0 {
  margin: 10px auto;
}

.c0 div {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  min-height: 40px;
  z-index: initial;
  font-size: 16px;
  background: #ffffff;
  caret-color: auto;
  border: 1px solid #C9D3DF;
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  -webkit-transition: opacity 0.2s ease-in-out,border-color 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out,border-color 0.2s ease-in-out;
}

.c0 div div:hover {
  border-left: 1px solid #738094;
  border-right: 1px solid #738094;
  border-top: 1px solid #738094;
  border-bottom: 1px solid #738094;
}

.c0 div input {
  height: 40px;
  box-sizing: border-box;
  width: 100%;
  padding: 0 14px;
  border-radius: 2px;
  line-height: 32px;
  font-size: 16px;
  border: none;
  color: #323b4d;
}

.c0 div input:focus {
  outline: none;
}

<div>
  <div
    class="c0"
  >
    <div>
      <input
        id="sc-iCfMLu c0"
        placeholder="Type here"
      />
    </div>
  </div>
</div>
`;
